File: algorithms/1_easy/C/03340_check_balanced_string/solution.py
class Solution:
    def isBalanced(self, num: str) -> bool:
        balance = 0

        for (i, ch) in enumerate(num):
            if (i & 1) == 1:
                balance += ord(ch) - ord('0')
            else:
                balance -= ord(ch) - ord('0')

        return balance == 0

File: algorithms/1_easy/C/03340_check_balanced_string/solution.rs
impl Solution {
    pub fn is_balanced(num: String) -> bool {
        let mut balance = 0;

        for (i, &byte) in num.as_bytes().iter().enumerate() {
            if (i & 1) == 1 {
                balance += (byte - b'0') as i32;
            } else {
                balance -= (byte - b'0') as i32;
            }
        }

        balance == 0
    }
}
